### PR TITLE
Add Gaussian Pulsed Plane Wave field

### PR DESCRIPTION
**Pull request by steindev**
- [x] rebase and merge after #26

Add Gaussian pulsed plane wave field. In principle this is a copy-paste-replace implementation of the `CosSquarePulse`.

An implementation of the `generic_spectrum` computation using pre-calculated formulas would be good, but can be added later.

### CODE REVIEW

**Comment by szabo137** [comment]
Please format your code right by hitting 

```julia
# install dependencies
julia --project=.formatting -e 'import Pkg; Pkg.instantiate()'
# format all documents
julia --project=.formatting .formatting/format_all.jl
```
locally 😺

File: src/pulses/gaussian.jl [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+"""
+struct GaussianPulse{M<:QEDbase.AbstractFourMomentum,T<:Real} <:
+       AbstractPulsedPlaneWaveField
+    mom::M
+    pulse_width::T
+end
+
+@inline function _unsafe_gaussian_envelope(phi, dphi)
+    return exp(-(phi / dphi)^2)
```
Comment (szabo137) [inline src/pulses/gaussian.jl:36]: I think there should be a two in the denominator. Then the `dphi^2` is identical to the variance of the Gaussian.

**Review by szabo137 — COMMENTED** [review]
Thanks for implementing this. I have just some small comments. 

Regarding the analytical formulas of the generic spectrum, could you please open a respective issue and remind us to add it later?

File: src/pulses/gaussian.jl (hunk 1 above)
Comment (steindev) [reply src/pulses/gaussian.jl:36]: Just a matter of taste...But will do.

**Comment by szabo137** [comment]
Since #26 is merged, please rebase to `dev` and force push.

File: src/pulses/gaussian.jl [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+####
+# interface functions
+####
+
+reference_momentum(pulse::GaussianPulse) = pulse.mom
+
+function domain(pulse::GaussianPulse)
+    dphi = pulse.pulse_width
+    return Interval(-7.0 * dphi, 7.0 * dphi)
```
[the remark this answers is not in the document]
Comment (szabo137) [reply src/pulses/gaussian.jl]: Tests need to be updated as well ;-)

**Comment by szabo137** [comment]
We are currently in the middle of a refactoring of `QED.jl` (see https://github.com/QEDjl-project/QED.jl/issues/35 for details). Therefore we prioritized #29 over this one. If you want to proceed, I suggest merging the current dev into your branch and fixing the merge conflicts. Then force-pushing again. Please be aware of the `QEDbase` namespace barrier introduced in #29 (see the issue on restructuring above for clarification or just ask here)

**Comment by szabo137** [comment]
@steindev The refactoring is finished successfully. Please rebase and force push to proceed with this one.

**Comment by steindev** [comment]
@szabo137 PR is rebased and ready for review/merge.

**Review by szabo137 — APPROVED** [review]
Excellent! Looks good for me.

**Pull request merged**